coriolis/tests/minion_manager/rpc/test_server.py

```diff
@@ -0,0 +1,201 @@
+        expected_result = expect.get("result")
+
+        minion_pool = testutils.DictToObject(config.get("minion_pool"))
+        action_instances = config.get("action_instances")
+
+        args = [
+            mock.sentinel.context,
+            minion_pool,
+            mock.sentinel.action_id,
+            action_instances,
+            mock.sentinel.subflow_name,
+        ]
+
+        if expected_exception:
+            exception_type = getattr(exception, expected_exception)
+            self.assertRaises(
+                exception_type,
+                self.server._make_minion_machine_allocation_subflow_for_action,
+                *args)
+            return
+
+        result = self.server\
+            ._make_minion_machine_allocation_subflow_for_action(
+                *args)
+
+        mappings = expected_result.get("mappings")
+
+        num_new_machines = list(mappings.values()).count("new_machine")
+
+        # db_api.add_minion_machine is called once for each new machine
+        self.assertEqual(
+            num_new_machines,
+            mock_add_minion_machine.call_count)
+
+        num_non_new_machines = len(mappings) - num_new_machines
+        if num_non_new_machines:
+            # db_api.set_minion_machines_allocation_statuses is called once
+            # with the non-new machines
+            mock_set_minion_machines_allocation_statuses\
+                .assert_called_once_with(
+                    mock.sentinel.context,
+                    list(mappings.values())[:num_non_new_machines],
+                    mock.sentinel.action_id,
+                    constants.MINION_MACHINE_STATUS_RESERVED,
+                    refresh_allocation_time=True)
+
+        self.assertEqual(
+            mappings,
+            result.get("action_instance_minion_allocation_mappings"))
+
+    @mock.patch.object(db_api, "set_minion_machine_allocation_status")
+    @mock.patch.object(
+        server.MinionManagerServerEndpoint,
+        "_add_minion_pool_event")
+    @mock.patch.object(
+        server.MinionManagerServerEndpoint,
+        "_get_minion_pool")
+    @ddt.file_data(
+        "data/get_minion_pool_refresh_flow.yaml"
+    )
+    @ddt.unpack
+    def test_get_minion_pool_refresh_flow(
+            self,
+            mock_get_minion_pool,
+            mock_add_minion_pool_event,
+            mock_set_minion_machine_allocation_status,
+            config,
+            expect,
+    ):
+        minion_pool = testutils.DictToObject(config.get("minion_pool"), {})
+        expected_result = expect.get("result", {})
+        expect_exception = expect.get("exception")
+        exptected_flow_tasks = expected_result.get("flow_tasks", [])
+        expected_db_calls = expected_result.get("db_calls", {})
+        expected_db_calls_include = expected_db_calls.get("include", [])
+        expected_db_calls_exclude = expected_db_calls.get("exclude", [])
+
+        mock_get_minion_pool.return_value = minion_pool
```

Comment:
You set this return value, but never actually use it, because you never pass `requery` argument (you only test by passing `requery=False`). Please also pass that and check whether this method is called accordingly.